src/Chapter3.hs

```diff
@@ -480,6 +514,36 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data Castle = MkCastle { castleName :: String }
+
+data Wall = Wall
+
+data ThridObject = Church | Library
+
+data House = MkHouse { housePeople :: Int }
+
+data City = MkCity
+  { castle :: Maybe Castle
+  , wall :: Maybe Wall
```

Comment:
This is the correct data type, that can represent what we have in requirements 👍🏼 
However, this also allows having some states that should not ideally exist: like having walls without any castle at all. With this representation, this is possible. However, we actually can restrict this on type-level by providing more representative types 🙂 This is actually the greatest power of Haskell and type system 😍 

This pattern is called "Make illegal states unrepresentable", and you can read more about it here:

* https://kowainik.github.io/posts/haskell-mini-patterns#make-illegal-states-unrepresentable

Reply:
:+1: